airbyte-oauth/src/main/java/io/airbyte/oauth/flows/SquareOAuthFlow.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.oauth.flows;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import io.airbyte.config.persistence.ConfigRepository;
+import io.airbyte.oauth.BaseOAuthFlow;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URLDecoder;
+import java.net.http.HttpClient;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Supplier;
+import org.apache.http.client.utils.URIBuilder;
+
+public class SquareOAuthFlow extends BaseOAuthFlow {
+
+  private static final String SCOPE_VALUE =
+      "ITEMS_READ+CUSTOMERS_WRITE+MERCHANT_PROFILE_READ+EMPLOYEES_READ+PAYMENTS_READ+CUSTOMERS_READ+TIMECARDS_READ+ORDERS_READ";
+  private static final String AUTHORIZE_URL = "https://connect.squareup.com/oauth2/authorize";
+  private static final String ACCESS_TOKEN_URL = "https://connect.squareup.com/oauth2/token";
+
+  public SquareOAuthFlow(ConfigRepository configRepository) {
+    super(configRepository);
+  }
+
+  @VisibleForTesting
+  public SquareOAuthFlow(ConfigRepository configRepository,
+                         HttpClient httpClient,
+                         Supplier<String> stateSupplier) {
+    super(configRepository, httpClient, stateSupplier);
+  }
+
+  @Override
+  protected String formatConsentUrl(UUID definitionId, String clientId, String redirectUrl)
+      throws IOException {
+    try {
+      // Need to have decoded format, otherwice square fails saying that scope is incorrect
+      return URLDecoder.decode(new URIBuilder(AUTHORIZE_URL)
+          .addParameter("client_id", clientId)
+          // .addParameter("redirect_uri", redirectUrl)
```

Comment:
please, remove commented line if not needed

Reply:
Updated, thanks!